infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServerNode.java

```diff
@@ -136,30 +139,36 @@ public void close() {
 
         // A executor service to create the shutdown threads
         // plus name the threads correctly.
-        final ExecutorService shutdownService = Executors.newFixedThreadPool(serverMap.size(),
-                new ServerThreadFactory("CorfuServer-shutdown-",
-                        new ServerThreadFactory.ExceptionHandler()));
+        final ExecutorService shutdownService = Executors.newFixedThreadPool(
+                serverMap.size(),
+                new ServerThreadFactory("CorfuServer-shutdown-", new ExceptionHandler())
+        );
```

Comment:
Honestly having this seems like an overkill. Whoever wrote this probably was trying to use the same design as the system's shutdown hook, but that is a little different that what we are doing during shutdown. Shutdown hooks are system wide and have multiple entry points. But since these are completely controlled by us, don't see a strong reason for parallelism here. I would just execute it in a single thread.

Reply:
Agree, but I wouldn't like to touch it now

Reply:
fixed